stl/inc/xlocale

```diff
@@ -74,12 +74,18 @@ public:
 
     class _CRTIMP2_PURE_IMPORT id { // identifier stamp, unique for each distinct kind of facet
     public:
-        __CLR_OR_THIS_CALL id(size_t _Val = 0) : _Id(_Val) {}
+        template <int = 0> // TRANSITION, ABI
+        id() noexcept /* strengthened */ {}
+
+#ifdef _CRTBLD // TRANSITION, ABI: preserved for binary compatibility
+        __CLR_OR_THIS_CALL id(size_t _Val) : _Id(_Val) {}
+#endif // defined(_CRTBLD)

```

Comment:
Unfortunately, this causes the following export to disappear, as revealed by `dumpbin /exports`, the entrance into knowledge of all existing things and all obscure secrets:

```
??_Fid@locale@std@@QEAAXXZ = ??_Fid@locale@std@@QEAAXXZ (public: void __cdecl std::locale::id::`default constructor closure'(void))
```

I believe that we need to keep the default arg for `_CRTBLD`.